### PR TITLE
[CLOUDTRUST-5555] Update EventReporterModule to accept a kafka-client producer

### PR DESCRIPTION
**Pull request by jonas-tr**

### CODE REVIEW

**Comment by jonas-tr** [comment]
I wanted to have a review of what I did for now before implementing it in each service.

To see an example of this being used: https://bitbucket.svc.elca.ch/projects/CLOUDTRUST/repos/back-accreditations-service/pull-requests/44/overview

File: security/authorizationreloader.go [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+	kafkaUniverse   *kafkauniverse.KafkaUniverse
+	kafkaConsumerID string
+}
+
+type AuthorizationRelaoder interface {
+	StartConsumers()
+	Close()
+}
+
+func NewAuthorizationReloader(id string, authManager AuthorizationManager, logger log.Logger, ctx context.Context, v *viper.Viper, idGenerator idgenerator.IDGenerator, logEventRate int64) AuthorizationRelaoder {
```
Comment (fperot74) [inline security/authorizationreloader.go:24]: Would it be possible not to link this module with kafka-client?
We don't want to have multiple "KafkaUniverse" instances in a single application. May be we could have a way to link a given consumer of the KafkaUniverse to a callback function (something like func OnReceivedContent(byte[] content) error).
This callback function would be the entry point in the AuthorizationManager. This way, we don't need an authorization reloader, just a callback in AuthorizationManager and have no link with Kafka

kafkaUniverse.GetConsumer(id).
		...
		SetHandler(authManager.OnReceivedContent)

**Review by fperot74 — CHANGES_REQUESTED** [review]

File: security/authorizationreloader.go (hunk 1 above)
Comment (jonas-tr) [reply security/authorizationreloader.go:24]: Okay so if I understand correctly you suggest to only modify the kafka-client not the common-service.

Comment (fperot74) [reply security/authorizationreloader.go:24]: I suggest to modify:
- the kafka-client library to add an easy way of launching KafkaUniverse when you only need to run one consumer providing a basic consumer...
- the common-service library to be able to update the AuthorizationManager providing a simple input (probably of type []byte)

Comment (basbeu) [reply security/authorizationreloader.go:24]: I agree with Francis !

**Review by basbeu — CHANGES_REQUESTED** [review]

**Comment by jonas-tr** [comment]
I don't think any changes are needed in common-service

File: events/auditeventreporter.go [hunk 2]
```diff
@@ -38,8 +39,7 @@ func (e *auditEventsReporterModule) ReportEvent(ctx context.Context, event Event
 		key = "DEFAULT-KEY"
 	}
 
-	msg := &sarama.ProducerMessage{Topic: e.topic, Key: sarama.StringEncoder(key), Value: sarama.StringEncoder(base64Event)}
-	_, _, err := e.producer.SendMessage(msg)
+	err := e.producer.SendMessageBytes([]byte(base64Event))
```
Comment (basbeu) [inline events/auditeventreporter.go:42]: Very important here, you should keep the partition key ! 

Then you should used SendPartitionedMessageBytes methods from  kafka-client

**Review by basbeu — CHANGES_REQUESTED** [review]

**Review by basbeu — APPROVED** [review]

**Review by fperot74 — APPROVED** [review]
Approve... but reword your commit... This change is not related to authorization reloader

**Pull request merged**